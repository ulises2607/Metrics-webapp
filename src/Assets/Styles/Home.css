.home-container {
  display: flex;
  flex-wrap: wrap;
  height: 100%;
  overflow-y: scroll;
  width: 100%;
  scrollbar-width: none;
  scrollbar-color: transparent transparent;
  overflow-x: hidden;
}

.item {
  border: 1px solid #ccc;
  height: 180px;
  display: flex;
  justify-content: center;
  align-items: center;
}

.divider {
  border: 1px solid #ccc;
  height: 1px;
  width: 100%;
  background-color: #ccc;
}

.dark {
  background-color: #db4681;
}

.light {
  background-color: #cf4278;
}
