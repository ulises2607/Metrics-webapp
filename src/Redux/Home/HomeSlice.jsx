import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
import { SOUTH_AMERICA } from "../../database/database";

const initialState = {
    home: SOUTH_AMERICA,
    detail: [],
    error: null,
}

const getData = async (longitude, latitude) => {
    try {
        const response = await fetch(`https://api.openweathermap.org/data/2.5/air_pollution/forecast?lat=${latitude}&lon=${longitude}&appid=3e341f1094147b1dd3b25d397b2c892e`);
        const data = await response.json();
        return data.list[0].components;
    } catch (error) {
        console.log(error, 'error en el getData');
    }
}

export const fetchHome = createAsyncThunk('home/fetchHome', async ( country ) => {
    const name = ['Salta', 'BuenosAires', 'Cordoba', 'Mendoza', 'Rosario', 'Tucuman'];
    const final = [];

    await Promise.all(
        name.map(async (element) => {
            try {
                const response = await fetch(`https://api.openweathermap.org/geo/1.0/direct?q=${element}&appid=3e341f1094147b1dd3b25d397b2c892e`);
                const data = await response.json();
                const { lat, lon } = data[0];
                const data2 = await getData(lon, lat);
                final.push(data2)
                // final.push(data[0]);
            } catch (error) {
                console.log(error, 'error en el fetchhome');
            }
        })
    );

    return final;
});





const homeSlice = createSlice({
    name: 'home',
    initialState,
    reducers: {
        setHome: (state, action) => {
            state.home = action.payload
        }
    },
    extraReducers: (builder) => {
        builder
        .addCase(fetchHome.pending, (state) => {
            state.error = undefined;
          })
        .addCase(fetchHome.fulfilled, (state, action) => {
            state.detail = action.payload;
            const timestamp = 1694631600;

            const date = new Date(timestamp * 1000);

            const fechaLegible = date.toLocaleString();

            console.log(fechaLegible);
            console.log(state.home);
        })
    }
})

export const { setHome } = homeSlice.actions

export default homeSlice
