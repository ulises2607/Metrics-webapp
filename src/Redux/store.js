import { configureStore } from "@reduxjs/toolkit";
import homeSlice from "./Home/HomeSlice";

const store = configureStore({
    reducer: {
        home: homeSlice.reducer,
    }
});

export default store;