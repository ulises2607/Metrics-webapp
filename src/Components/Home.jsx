import React from 'react'
import '../Assets/Styles/Home.css'

const Home = () => {
  return (
    <div className='home-container'>
      <div className="item">Argentina</div>
      <div className="item">Bolivia</div>
      <div className="item">Peru</div>
      <div className="item">Chile</div>
      <div className="item">Uruguay</div>
      <div className="item">Brasil</div>
    </div>
  )
}

export default Home