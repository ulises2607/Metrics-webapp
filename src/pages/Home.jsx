import '../Assets/Styles/Home.css';
import React from 'react';
import { useSelector } from 'react-redux';
import Header from '../Components/Header';
import HomeCards from '../Components/HomeCards';

const Home = () => {
  const { home } = useSelector((state) => state.home);
  return (
    <>
      <div className="header-container">
        <Header />
      </div>
      <div className="home-container">
        {home.map((item) => (
          <HomeCards key={item.nombre} item={item} />
        ))}
      </div>
    </>
  );
};

export default Home;
