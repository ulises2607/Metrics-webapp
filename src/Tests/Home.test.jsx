import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Home from '../pages/Home';
import store from '../Redux/store';

test('Snapshot of the Home component', () => {
  const component = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
