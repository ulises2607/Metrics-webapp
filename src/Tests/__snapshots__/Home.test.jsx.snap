// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of the Home component 1`] = `
Array [
  <div
    className="header-container"
  >
    <div
      className="header-container"
    >
      <div
        className="img-header-container"
      >
        <img
          alt="logo"
          className="logo-header"
          src="pngegg.png"
        />
      </div>
      <div
        className="title-header-container"
      >
        <h1
          className="title-header"
        >
          South America air polution
        </h1>
      </div>
    </div>
  </div>,
  <div
    className="divider"
  />,
  <div
    className="home-container"
  >
    <div
      className="homeCard-container dark"
    >
      <div
        className="info-card"
      >
        <img
          alt="GIF"
          className="location-img"
          src="IMG-20230915-WA0004.jpg"
        />
        <h2>
          Argentina
        </h2>
      </div>
      <a
        href="/details/Argentina"
        onClick={[Function]}
      >
        <img
          alt="GIF"
          className="arrow-img"
          src="arrow-right.png"
        />
      </a>
    </div>
    <div
      className="homeCard-container light"
    >
      <div
        className="info-card"
      >
        <img
          alt="GIF"
          className="location-img"
          src="IMG-20230915-WA0004.jpg"
        />
        <h2>
          Bolivia
        </h2>
      </div>
      <a
        href="/details/Bolivia"
        onClick={[Function]}
      >
        <img
          alt="GIF"
          className="arrow-img"
          src="arrow-right.png"
        />
      </a>
    </div>
    <div
      className="homeCard-container light"
    >
      <div
        className="info-card"
      >
        <img
          alt="GIF"
          className="location-img"
          src="IMG-20230915-WA0004.jpg"
        />
        <h2>
          Brazil
        </h2>
      </div>
      <a
        href="/details/Brazil"
        onClick={[Function]}
      >
        <img
          alt="GIF"
          className="arrow-img"
          src="arrow-right.png"
        />
      </a>
    </div>
    <div
      className="homeCard-container dark"
    >
      <div
        className="info-card"
      >
        <img
          alt="GIF"
          className="location-img"
          src="IMG-20230915-WA0004.jpg"
        />
        <h2>
          Chile
        </h2>
      </div>
      <a
        href="/details/Chile"
        onClick={[Function]}
      >
        <img
          alt="GIF"
          className="arrow-img"
          src="arrow-right.png"
        />
      </a>
    </div>
    <div
      className="homeCard-container dark"
    >
      <div
        className="info-card"
      >
        <img
          alt="GIF"
          className="location-img"
          src="IMG-20230915-WA0004.jpg"
        />
        <h2>
          Colombia
        </h2>
      </div>
      <a
        href="/details/Colombia"
        onClick={[Function]}
      >
        <img
          alt="GIF"
          className="arrow-img"
          src="arrow-right.png"
        />
      </a>
    </div>
    <div
      className="homeCard-container light"
    >
      <div
        className="info-card"
      >
        <img
          alt="GIF"
          className="location-img"
          src="IMG-20230915-WA0004.jpg"
        />
        <h2>
          Ecuador
        </h2>
      </div>
      <a
        href="/details/Ecuador"
        onClick={[Function]}
      >
        <img
          alt="GIF"
          className="arrow-img"
          src="arrow-right.png"
        />
      </a>
    </div>
    <div
      className="homeCard-container light"
    >
      <div
        className="info-card"
      >
        <img
          alt="GIF"
          className="location-img"
          src="IMG-20230915-WA0004.jpg"
        />
        <h2>
          Guyana
        </h2>
      </div>
      <a
        href="/details/Guyana"
        onClick={[Function]}
      >
        <img
          alt="GIF"
          className="arrow-img"
          src="arrow-right.png"
        />
      </a>
    </div>
    <div
      className="homeCard-container dark"
    >
      <div
        className="info-card"
      >
        <img
          alt="GIF"
          className="location-img"
          src="IMG-20230915-WA0004.jpg"
        />
        <h2>
          Paraguay
        </h2>
      </div>
      <a
        href="/details/Paraguay"
        onClick={[Function]}
      >
        <img
          alt="GIF"
          className="arrow-img"
          src="arrow-right.png"
        />
      </a>
    </div>
    <div
      className="homeCard-container dark"
    >
      <div
        className="info-card"
      >
        <img
          alt="GIF"
          className="location-img"
          src="IMG-20230915-WA0004.jpg"
        />
        <h2>
          Peru
        </h2>
      </div>
      <a
        href="/details/Peru"
        onClick={[Function]}
      >
        <img
          alt="GIF"
          className="arrow-img"
          src="arrow-right.png"
        />
      </a>
    </div>
    <div
      className="homeCard-container light"
    >
      <div
        className="info-card"
      >
        <img
          alt="GIF"
          className="location-img"
          src="IMG-20230915-WA0004.jpg"
        />
        <h2>
          Suriname
        </h2>
      </div>
      <a
        href="/details/Suriname"
        onClick={[Function]}
      >
        <img
          alt="GIF"
          className="arrow-img"
          src="arrow-right.png"
        />
      </a>
    </div>
    <div
      className="homeCard-container light"
    >
      <div
        className="info-card"
      >
        <img
          alt="GIF"
          className="location-img"
          src="IMG-20230915-WA0004.jpg"
        />
        <h2>
          Uruguay
        </h2>
      </div>
      <a
        href="/details/Uruguay"
        onClick={[Function]}
      >
        <img
          alt="GIF"
          className="arrow-img"
          src="arrow-right.png"
        />
      </a>
    </div>
    <div
      className="homeCard-container dark"
    >
      <div
        className="info-card"
      >
        <img
          alt="GIF"
          className="location-img"
          src="IMG-20230915-WA0004.jpg"
        />
        <h2>
          Venezuela
        </h2>
      </div>
      <a
        href="/details/Venezuela"
        onClick={[Function]}
      >
        <img
          alt="GIF"
          className="arrow-img"
          src="arrow-right.png"
        />
      </a>
    </div>
  </div>,
]
`;
